### PR TITLE
Repay liquidity

### PR DESCRIPTION
**Pull request by yisalk**

### CODE REVIEW

File: src/hooks/useRepayLoanHandler.tsx [hunk 1]
```diff
@@ -1,24 +1,129 @@
...
+        address,
+        PosManager.abi,
+        accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+      )
+      if (_posManager) {
+        setPosManager(_posManager)
+      }
+    }
+    fetchContract()
+  }, [provider])
```
Comment (TBomberman) [inline src/hooks/useRepayLoanHandler.tsx:54]: Doesn't the provider change only when the user connects with metamask? What happens if the user does not gone to the repay loan page yet while he is already connected? When will these contracts get set?

Comment (TBomberman) [reply src/hooks/useRepayLoanHandler.tsx:54]: Provider changes when page loads. Because of WalletContext.

**Review by Pochetes — COMMENTED** [review]
Again, what happened here? I shouldn't be trying to guess these PRs

**Comment by TBomberman** [comment]
The goal of this PR is to get repay liquidity page to the state that we can call the repay function of the contract with the button. There's also changes to valldiation and slider behavior.

**Review by dam417 — APPROVED** [review]
Approving this PR so we can move to next step. Once smart contract is unit tested we can resolve any issues we find here since we'll know issues are here. The standard for now is as long as it's not crashing the front end it's fine.

**Comment by TBomberman** [comment]
Made repay sliders set to the actual values.
Prepare repay handler for repaying by loading pool and posmgr contracts.
Create actual function for calling the repay contract function.

**Pull request merged**